src/change_stream/event.rs

```diff
@@ -142,13 +147,11 @@ pub enum OperationType {
 }
 
 /// Identifies the collection or database on which an event occurred.
-#[derive(Deserialize, Debug)]
-#[serde(untagged)]
-#[non_exhaustive]
-pub enum ChangeStreamEventSource {
-    /// The [`Namespace`] containing the database and collection in which the change occurred.
-    Namespace(Namespace),
+#[derive(Deserialize, Debug, PartialEq, Eq)]
+pub struct ChangeStreamEventSource {
```

Comment:
is it possible this could change in the future? if so, we should keep it `non_exhaustive`

Reply:
Good call, done.